x-pack/plugins/rule_registry/server/rule_data_client/rule_data_client.ts

```diff
@@ -95,62 +115,87 @@ export class RuleDataClient implements IRuleDataClient {
   }
 
   public getWriter(options: { namespace?: string } = {}): IRuleDataWriter {
-    const { indexInfo, resourceInstaller } = this.options;
-
     const namespace = options.namespace || 'default';
+    const cachedWriter = this.writerCache.get(namespace);
+
+    // There is no cached writer, so we'll install / update the namespace specific resources now.
+    if (!cachedWriter) {
+      const writerForNamespace = this.initializeWriter(namespace);
+      this.writerCache.set(namespace, writerForNamespace);
+      return writerForNamespace;
+    } else {
+      return cachedWriter;
+    }
+  }
+
+  private initializeWriter(namespace: string): IRuleDataWriter {
```

Comment:
Do we want to use `_` for private methods?

Reply:
Same as above, I think it's nicer to let `private` denote the visibility. I know `_isWriteEnabled` skews the consistency slightly although that was just because of the getters / setters conflicting with the public method and I didn't want to use a name that sounded confusing. Let me think on the naming there, there's possibly a way to stop the conflict.